rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -25,6 +25,54 @@
 #include <rosidl_typesupport_opensplice_cpp/message_type_support.h>
 #include <rosidl_typesupport_opensplice_cpp/service_type_support.h>
 
+inline std::string
+_create_type_name(
+  const message_type_support_callbacks_t * callbacks,
+  const std::string & sep)
+{
+  return std::string(callbacks->package_name) +
+         "::" + sep + "::dds_::" + callbacks->message_name + "_";
+}
+
+// *INDENT-OFF* (prevent uncrustify from using four space indention here)
+#define TRY_PLACEMENT_NEW(Destination, BufferForNew, FailureAction, Type, ...) try { \
```

Comment:
I explicitly avoided this, since the implementations do not share headers other than from `rmw`. I don't think `rmw` is an appropriate place to put these macros. I'm open to suggestion on how to share them (with `rmw_opensplice` as well).